### PR TITLE
[Kettle] Add some docs on Kettle and PubSub

### PR DESCRIPTION
**Pull request by MushuEE**
Describe auth needs to PubSub changes that I made this morning. 

For reference the failure was as follows:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/dist-packages/google/cloud/pubsub_v1/_gapic.py", line 40, in <lambda>
    fx = lambda self, *a, **kw: wrapped_fx(self.api, *a, **kw)  # noqa
  File "/usr/local/lib/python3.6/dist-packages/google/pubsub_v1/services/subscriber/client.py", line 1012, in pull
    response = rpc(request, retry=retry, timeout=timeout, metadata=metadata,)
  File "/usr/local/lib/python3.6/dist-packages/google/api_core/gapic_v1/method.py", line 145, in __call__
    return wrapped_func(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/google/api_core/retry.py", line 286, in retry_wrapped_func
    on_error=on_error,
  File "/usr/local/lib/python3.6/dist-packages/google/api_core/retry.py", line 184, in retry_target
    return target()
  File "/usr/local/lib/python3.6/dist-packages/google/api_core/grpc_helpers.py", line 59, in error_remapped_callable
    six.raise_from(exceptions.from_grpc_error(exc), exc)
  File "<string>", line 3, in raise_from
google.api_core.exceptions.PermissionDenied: 403 User not authorized to perform this action.
```
/area kettle
/cc @spiffxp @amwat

### CODE REVIEW

**Comment by spiffxp** [comment]
```
==================== Test output for //hack:verify-spelling:
Validating spelling...
./kettle/README.md:114:19: "permisions" is a misspelling of "permission"
ERROR: bad spelling, fix with hack/update-spelling.sh
```

File: kettle/README.md [hunk 1]
```diff
@@ -100,10 +100,25 @@ gs://<bucket path>: #bucket url
...
+Kettle `stream.py` leverages Google Cloud [PubSub] to alert on GCS changes. These events are tied to the `gcs-changes` Topic in the `kubernetes-jenkins` project where Prow job artifacts are collated. Each time an artifact is finalized, a PubSub event is triggered and Kettle collects job information when it sees a resource uploaded called `finished.json` (indicating the build completed).
+
+[Subscriptions] are in Kuberenetes Jenkins Build - PubSub.
+- kettle
+- kettle-staging
+
+They are split so that the staging instance does not consume events aimed at production.
+
+### Auth
+For kettle to have permisions, kettle's user needs access. When updating or changing a [Subscription] make sure to add `kettle@k8s-gubernator.iam.gserviceaccount.com` as a `PubSub Editor`.
```
Comment (spiffxp) [inline kettle/README.md:114]: nit: gcloud example to copy-paste would be helpful

File: kettle/README.md [hunk 2]
```diff
@@ -100,10 +100,25 @@ gs://<bucket path>: #bucket url
 
 A [postsubmit job](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/test-infra/test-infra-trusted.yaml#L203-L210) runs that pushes Kettle on changes.
 
+# PubSub
+
+Kettle `stream.py` leverages Google Cloud [PubSub] to alert on GCS changes. These events are tied to the `gcs-changes` Topic in the `kubernetes-jenkins` project where Prow job artifacts are collated. Each time an artifact is finalized, a PubSub event is triggered and Kettle collects job information when it sees a resource uploaded called `finished.json` (indicating the build completed).
+
+[Subscriptions] are in Kuberenetes Jenkins Build - PubSub.
+- kettle
+- kettle-staging
```
Comment (spiffxp) [inline kettle/README.md:109]: Should the kettle subscription have the same filter as kettle-staging?
```yaml
# gcloud pubsub subscriptions list --project=kubernetes-jenkins
---
ackDeadlineSeconds: 10
expirationPolicy:
  ttl: 2678400s
messageRetentionDuration: 604800s
name: projects/kubernetes-jenkins/subscriptions/kettle
pushConfig: {}
topic: projects/kubernetes-jenkins/topics/gcs-changes
---
ackDeadlineSeconds: 10
expirationPolicy:
  ttl: 2678400s
filter: attributes.eventType = "OBJECT_FINALIZE"
messageRetentionDuration: 604800s
name: projects/kubernetes-jenkins/subscriptions/kettle-staging
pushConfig: {}
topic: projects/kubernetes-jenkins/topics/gcs-changes
```

File: kettle/README.md [hunk 3]
```diff
@@ -100,10 +100,25 @@ gs://<bucket path>: #bucket url
 
 A [postsubmit job](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/test-infra/test-infra-trusted.yaml#L203-L210) runs that pushes Kettle on changes.
 
+# PubSub
+
+Kettle `stream.py` leverages Google Cloud [PubSub] to alert on GCS changes. These events are tied to the `gcs-changes` Topic in the `kubernetes-jenkins` project where Prow job artifacts are collated. Each time an artifact is finalized, a PubSub event is triggered and Kettle collects job information when it sees a resource uploaded called `finished.json` (indicating the build completed).
```
Comment (spiffxp) [inline kettle/README.md:105]: Is the topic something we need to create?

The bucket and project are named the same, is this getting message for all buckets in the project, or just gs://kubernetes-jenkins?

Does this work for any of the other buckets listed in buckets.yaml or is this hardcoded?

**Review by spiffxp — COMMENTED** [review]
A few questions

File: kettle/README.md (hunk 1 above)
Comment (MushuEE) [reply kettle/README.md:114]: Sure, I used UI but I can try and find the command +1

File: kettle/README.md (hunk 2 above)
Comment (MushuEE) [reply kettle/README.md:109]: I added the filter as an experiment and wanted to make sure it worked before rebuilding prod. You can't add a filter to an existing subscription actually, so I really wanted to verify before destroying the existing one.

File: kettle/README.md (hunk 3 above)
Comment (MushuEE) [reply kettle/README.md:105]: This is hardcoded as part of the subscription. I am not 100% how the topic was created. I believe it is tied to the jobs in `kubernetes-jenkins` exclusively. We could setup a per-bucket event using the prefixes and fork the process across all buckets but would require topics and subscriptions created in each bucket + permission for kettle to access it, would be involved.

**Comment by spiffxp** [comment]
/test pull-test-infra-integration
#20498 (comment)

File: kettle/README.md (hunk 1 above)
Comment (spiffxp) [reply kettle/README.md:114]:
```sh
gcloud pubsub subscriptions add-iam-policy-binding \
  projects/kubernetes-jenkins/subscriptions/kettle-staging \
  --member=serviceAccount:kettle@k8s-gubernator.iam.gserviceaccount.com \
  --role=roles/pubsub.editor
```

File: kettle/README.md (hunk 2 above)
Comment (spiffxp) [reply kettle/README.md:109]: I guess my question would be: when we decide to setup subscriptions on other buckets (either adding more, or switching to a community owned bucket), should we use the filter going forward?

File: kettle/README.md (hunk 3 above)
Comment (spiffxp) [reply kettle/README.md:105]: based on https://cloud.google.com/storage/docs/reporting-changes

I think the topic may have been auto-created by this command:
```
gsutil notification create -t projects/kubernetes-jenkins/topics/gcs-changes -f json gs://kubernetes-jenkins
```

Comment (spiffxp) [reply kettle/README.md:105]: also maybe change first sentence to "alert on GCS changes in the `kubernetes-jenkins` bucket"

File: kettle/README.md (hunk 2 above)
Comment (MushuEE) [reply kettle/README.md:109]: Ah, yes we would want the filter. Really we would like a suffix filter if that is added as well.

File: kettle/README.md (hunk 1 above)
Comment (MushuEE) [reply kettle/README.md:114]: Added with the command to make the subscriptions.

File: kettle/README.md (hunk 3 above)
Comment (MushuEE) [reply kettle/README.md:105]: Done

**Comment by MushuEE** [comment]
@amwat able to LGTM?

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Pull request merged**